mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -129,34 +131,25 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
     /* 
         Send whatever message is currently in the chat input
     */
-    const sendChatInputMessage = async (input: string) => {
+    const sendChatInputMessage = async (input: string, messageIndex?: number) => {
         // Step 0: Reject the previous Ai generated code if they did not accept it
         rejectAICode()
 
         // Step 1: Add the user's message to the chat history
         const newChatHistoryManager = getDuplicateChatHistoryManager()
-        newChatHistoryManager.addChatInputMessage(input)
+        if (messageIndex !== undefined) {
+            newChatHistoryManager.updateMessageAtIndex(messageIndex, input)
+        } else {
+            newChatHistoryManager.addChatInputMessage(input)
+        }
 
         // Step 2: Send the message to the AI
         const aiMessage = await _sendMessageToOpenAI(newChatHistoryManager)
```

Comment:
I know I wrote this code, but now reading it, I think a better name for this function would be `_sendMessageAndSaveResponse` just so its clear that we're also saving it to the chat history manager. Thoughts?

Reply:
Done. Your right, this was more apt name when most of the steps lived in `sendChatInputMessage`, but things have been moving out of that function at a pretty fast pace.